parts/linux/cloud-init/artifacts/cse_main.sh

```diff
@@ -39,448 +37,501 @@ python3 /opt/azure/containers/provision_redact_cloud_config.py \
+            # logs_to_events should not be run on & commands
+            if [ "${ENABLE_UNATTENDED_UPGRADES}" = "true" ]; then
+                UU_CONFIG_DIR="/etc/apt/apt.conf.d/99periodic"
+                mkdir -p "$(dirname "${UU_CONFIG_DIR}")"
+                touch "${UU_CONFIG_DIR}"
+                chmod 0644 "${UU_CONFIG_DIR}"
+                echo 'APT::Periodic::Update-Package-Lists "1";' >> "${UU_CONFIG_DIR}"
+                echo 'APT::Periodic::Unattended-Upgrade "1";' >> "${UU_CONFIG_DIR}"
+                systemctl unmask apt-daily.service apt-daily-upgrade.service
+                systemctl enable apt-daily.service apt-daily-upgrade.service
+                systemctl enable apt-daily.timer apt-daily-upgrade.timer
+                systemctl restart --no-block apt-daily.timer apt-daily-upgrade.timer
+                # this is the DOWNLOAD service
+                # meaning we are wasting IO without even triggering an upgrade
+                # -________________-
+                systemctl restart --no-block apt-daily.service
+
+            fi
+            aptmarkWALinuxAgent unhold &
+        elif isMarinerOrAzureLinux "$OS"; then
+            if [ "${ENABLE_UNATTENDED_UPGRADES}" = "true" ]; then
+                if [ "${IS_KATA}" = "true" ]; then
+                    # Currently kata packages must be updated as a unit (including the kernel which requires a reboot). This can
+                    # only be done reliably via image updates as of now so never enable automatic updates.
+                    echo 'EnableUnattendedUpgrade is not supported by kata images, will not be enabled'
+                else
+                    # By default the dnf-automatic is service is notify only in Mariner.
+                    # Enable the automatic install timer and the check-restart timer.
+                    # Stop the notify only dnf timer since we've enabled the auto install one.
+                    # systemctlDisableAndStop adds .service to the end which doesn't work on timers.
+                    systemctl disable dnf-automatic-notifyonly.timer
+                    systemctl stop dnf-automatic-notifyonly.timer
+                    # At 6:00:00 UTC (1 hour random fuzz) download and install package updates.
+                    systemctl unmask dnf-automatic-install.service || exit $ERR_SYSTEMCTL_START_FAIL
+                    systemctl unmask dnf-automatic-install.timer || exit $ERR_SYSTEMCTL_START_FAIL
+                    systemctlEnableAndStart dnf-automatic-install.timer 30 || exit $ERR_SYSTEMCTL_START_FAIL
+                    # The check-restart service which will inform kured of required restarts should already be running
+                fi
+            fi
+        fi
+    fi
+}
+
+# The provisioning is split into two stages to support VHD image creation workflows:
+#
+# Stage 1: Base image preparation
+#   - Installs and configures all required components (kubelet, containerd, etc.)
```

Comment:
Where / how can we ensure that users are aware that ALL RP controlled versioning / patching won't work with this approach ?  I'm also guessing Embargo support is out the window ? Users are on their own at this point when it comes to CVE/Patching.

Reply:
VHD still should behaves in the same way. So techincally, there should be no difference for applying patching on current VHD vs cached VHD